### ISSUE TITLE
I have find a bug for "git_status_list_entrycount()"

### ISSUE THREAD

**Issue by xiangism** [libgit2/libgit2#3417]
My OS is win7. This is my code
```
...
git_repository_open(&repo, "E:\\cpp_app\\BoostApp"); 
git_status_list *status;
git_status_options statusopt = GIT_STATUS_OPTIONS_INIT;
git_status_list_new(&status, repo, &statusopt);

int maxi = git_status_list_entrycount(status);
...
```
In this case, The value of maxi is right. But when I use "subst" command create the virtual disc, 
```subst i: e:\cpp_app\BoostApp```

And then, change the code at  open function
```
git_repository_open(&repo, "i:"); 
```
this time, the value of maxi is error, it looks like a random number , such as "36","1202", but it is right value is "0".

**Comment by ethomson** [comment]
It looks like you're not checking the return code of `git_repository_open`.  Are you sure that's even succeeding?

**Comment by carlosmn** [comment]
IIRC when you use the driver letter alone, such as with `"i:"` you ask Windows to open the *drive* rather than the filesystem on that drive. Does it work with `"i:\"` as the path?

And as Ed said, are you checking for errors? I would not expect opening that repository to work.

**Comment by xiangism** [comment]
@ethomson  @carlosmn  Neither use ```i:``` nor ```i:\```, the return code of ```git_repository_open()``` is 0, so it runs successfully. I used it with the implementation of add push commit fetch can be successful, only when there is a problem in  obtaining the status.

After ```git_status_list_new(&status, repo, &statusopt);``` , I celled "giterr_last()" to get the error message, this is the value:
```
error->klass = 2;
error->message = The global/xdg file 'ignore' doesn't exist: 绯荤粺鎵句笉鍒版寚瀹氱殑璺緞銆?
```
And the return code of ```git_status_list_new``` is "0".

**Comment by carlosmn** [comment]
Does `git` work in this situation? Are you running 'subst' and the libgit2-based command as the same user?

**Comment by ethomson** [comment]
Closed via #3425 - thanks for the bug report!
